switchlink/switchlink_main.c

```diff
@@ -288,6 +289,13 @@ static void nl_process_event_loop(void) {
 
     ret = select(num_fds, &read_fds, NULL, NULL, NULL);
     if (ret == -1) {
+      // Select system call being interrupted by a signal EINTR.
+      // This is not really an error condition, hence continue the loop.
+      if (errno == EINTR)
+        continue;
+
+      krnlmon_log_error("Received select error, errno :%d, meaning: %s",
+                         errno, strerror(errno));
       perror("pselect");
       return;
```

Comment:
What happens if we `return` here?

- Does it leave the system in an stable state?
- Is it the correct thing to do?
- If it causes krnlmon to shut down, should we use `krnlmon_log_critical` instead of `krnlmon_log_error`?

Reply:
Yes, this exits the krnlmon thread. Will log the an critical message.